replica/database.hh

```diff
@@ -710,9 +710,8 @@ public:
     // FIXME: in case a query is satisfied from a single memtable, avoid a copy
     using const_mutation_partition_ptr = std::unique_ptr<const mutation_partition>;
     using const_row_ptr = std::unique_ptr<const row>;
-    // FIXME: for supporting multiple compaction groups, this interface will need to return
-    // as many active sstables as there are groups.
-    memtable& active_memtable();
+    // Return all active memtables, where there will be one per compaction group
+    std::vector<std::reference_wrapper<memtable>> active_memtables();
```

Comment:
why don't we need a lw_shared_ptr<memtable> here like we do for compaction_group?

Reply:
Why not memtable*?

Reply:
Perhaps we can expose methods for operating the active memtables (retrieving statistics, etc) rather than exposing the memtables themselves.

Reply:
std::reference_wrapper is equivalent to a pointer, just takes longer to write.

memtable lifetime management must be careful, more careful than lw_shared_ptr which only ensures the object is alive but says nothing about whether it should be alive in the first place.

Reply:
> std::reference_wrapper is equivalent to a pointer, just takes longer to write.
> 
> memtable lifetime management must be careful, more careful than lw_shared_ptr which only ensures the object is alive but says nothing about whether it should be alive in the first place.

When I wrote that, the idea of using reference_wrapper was already abandoned in my mind. I think we need to change the interface. Instead of exposing memtable itself, expose what the current callers need from it.